### ISSUE TITLE
Add contact does not check for existing contact

### ISSUE THREAD

**Issue by clydelhui** [clydelhui/pe#5]
Multiple contacts are able to be added to applications.

![image.png](https://raw.githubusercontent.com/clydelhui/pe/main/files/95435944-c763-414a-884b-25bf45c51155.png)

The error message shows multiple contacts for the same application. However, the JSON data file only stores the latest contact added.

## Justification for severity
This bug may potentially affect users who key in the wrong index when adding contact. The command overwrites the data of previous add commands, only keeping the latest one. Hence, choosing the wrong index will cause the user to lose the contact of another entry. 
As indices are prone to user error, this may happen frequently, causing users to be unable to rely on the product to store contact details. 
Hence, the severity is high.

**Comment by nus-pe-script** [comment]
# Team's Response
First of all, thank you for pointing out this bug and I do agree that this is a functionality bug as checks were not made on whether there is an existing contact for the `add_contact` command.

However, the sentence below is **false**. 

> Multiple contacts are able to be added to applications.

Under **no** circumstances would an internship application have more than one contact added to it, as the newly added contact would **overwrite** the existing one according to the current behaviour. The tester may have misphrased the sentence, as he actually realized the behaviour, which is indicated by this sentence from the `Justification for severity` section.

> The command overwrites the data of previous add commands, only keeping the latest one.


Furthermore, the severity has to be downgraded to `severity.Medium` due to the following reasons:

To explain these reasons, it would be helpful to refer to the guide for bug severity labels given on the [module website](https://nus-cs2103-ay2223s2.github.io/website/schedule/week13/project.html#6-attend-the-practical-exam-during-the-lecture-on-fri-apr-14th).

![image.png](https://raw.githubusercontent.com/nus-cs2103-AY2223S2/pe/master/files/9bdf6a79-5be1-4234-94f5-7670f8800717.png)

1. The bug **does not** cause the product to be unusable for most users. **Only** the **contact** which consists of the phone number and email address is affected for a single application in the worst case, and the contact is **only one out of the many** attributes we allow the user to the enter for an internship application. **All** other data for the **same** internship application is **not** affected, and **all** data for **all** other internship applications remained **unaffected** by this bug. The justification that the user is unable to rely on the product to store contact details is **invalid** as **all** other contact details for the remaining applications are still **intact**.

2. While indices are prone to user error, this **does not** mean that most user will encounter this frequently. This **only** occurs **occasionally** so it **only** causes **occasional** inconvenience to the user. While I agree that most users are bound to enter a wrong index at some point, this **does not** indicate all of them will be doing so with the `add_contact` command, which further **invalidates** the point that this may occur frequently.

3. The index of a particular application is shown right beside the company name of the application and can be **easily** referenced by the user.

4. In the worst case where the user does make an index error, the resulting success message is displayed by the command, **not** an error message. The success message shows **both** the existing contact and the newly added contact of the particular application with the wrong index, and this message is shown for **5** seconds. In this 5 seconds, the user could take a look at the message displayed and **quickly** realize that there are **two** contacts present, which would **immediately** alert him/her that there is an existing contact. The user can then take a screenshot of this message to recover the previous phone number and email address that has now been overwritten, and while this may cause **some** inconvenience to the user, **no** data is lost in this case. Even under circumstances where the user has made an error, the product still provides a **reliable** way for the user to recover from the error, so once again it **cannot** be said that the user is unable to rely on the product to store contact details.

5. With a **possible** way for the user to recover from the error, which indicates that **there is** potential for the contact details to be recovered even after the user has entered a wrong index, this **does not** cause a major problem for the user.

6. The possibility for a user to enter a wrong index more than once is also **very low**, so it is **very unlikely** for the effect to snowball.

6. While this may not be true for all cases, our target users are primarily Computer Science undergraduates, **a lot** of which are sensitive to numbers, **further reducing** the possibility that a wrong index is entered.

In conclusion, as the justifications made for this bug which qualify it for `Severity.High` are **invalid**, it has to be downgraded to `Severity.Medium`.
# Items for the Tester to Verify
## :question: Issue severity

Team chose [`severity.Medium`]
Originally [`severity.High`]

- [ ] I disagree

**Reason for disagreement:** [replace this with your explanation]

<catcher-end-of-segment><hr>
